src/list-names.py

```diff
@@ -10,7 +10,9 @@ def main():
     )
     args = argparser.parse_args()
 
-    print(f"Now I need to process the records in {args.fastq}")
+    for l in args.fastq.readlines():
+        if l[0] == "@":
```

Comment:
This has the same issue as in the other exercise; it might crash on an empty line. But you can use `l.startswith('@')` instead, for a safe version.

```python
for l in (l for l in args.fasta if l.startswith('@')):
   print(l[1:].strip())
```

ok, that's not pretty, and that is because Python doesn't have a pretty pipe operator... but the comment on `startswith()` still applies.